packages/datadog-instrumentations/src/jest.js

```diff
@@ -72,6 +82,14 @@ function getWrappedEnvironment (BaseEnvironment) {
       this.testSuite = getTestSuitePath(context.testPath, rootDir)
       this.nameToParams = {}
       this.global._ddtrace = global._ddtrace
+
+      if (config.projectConfig && config.projectConfig.testEnvironmentOptions) { // newer versions
+        this._ddTestSessionId = config.projectConfig.testEnvironmentOptions._ddTestSessionId
```

Comment:
Also, why is this passed to the instrumentation in the first place? It should have no knowledge of tracing.

Reply:
this config is the only info that is passed to the worker thread that is spawned to run the tests

Reply:
Could this be done without passing the data to the instrumentation? The instrumentation should never know about tracing, otherwise we can never upstream the changes, and it's also bad practice generally speaking for the publisher to have to know about its subscriber.

Reply:
can you think of anything? We might be able to pass some environment variables to the worker threads that change the way dd-trace is initialised in them? I'm not sure if that's even possible and it sounds risky, as we may inadvertently affect processes we don't want to affect

Reply:
How does Jest connect them in the first place?

Reply:
the main processes starts a bunch of children processes and tells them which test files they need to run

Reply:
What I mean is how does the communication happen between the main process and the workers? How does the main process identifies what is done by which worker for example?

Reply:
they have cross process communication. The child reports its results to the parent: https://github.com/facebook/jest/blob/4def94b073cad300e99de378ba900e6ba9b7032f/packages/jest-worker/src/workers/processChild.ts#L71-L77

Reply:
Any way that we could link things from the information available there? If it's really not possible at all, any way to share something between the 2 processes that would be more realistic to add to Jest in the future? So basically something not related to Datadog.